demo/memdb_test.go

```diff
@@ -0,0 +1,19 @@
+//go:build memdb
+
+package main_test
+
+import (
+	"testing"
+
+	"github.com/drand/drand/chain"
+)
+
+func withTestDB() chain.StorageType {
+	return chain.MemDB
+}
+
+func withPgDSN(_ *testing.T) func() string {
```

Comment:
is this here because it has to exist in every impl? Not a big deal and I can't immediately think of another way that's not OTT, just wondering though!

Reply:
Yes, that's correct. Currently, the code uses the `withPgDSN` regardless of the selected database.
It would be a bit more hassle to make it adapt to the different engine types than to just add a stub function like this.

Reply:
I missed @AnomalRoil's comment, I'll blame Github on this.
To be more accurate, here's where the usage happens: https://github.com/drand/drand/blob/b204c4b5a57db95f63de297fc9212ab15278ee04/demo/demo_test.go#L34-L49

Specifically, the line 48.

A solution to prevent this would be to have the value of the `PgDSN` field set based on the value of `DBEngineType`. That, however, adds to the complexity of the test when reading it/keeping track of what happens.

I can make the change if desired.